test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,867 @@
+			targetNode := &workerRTNodes[0]
+			mm1.cpu = fmt.Sprintf("%d", len(available_node0_cpus)-2)
+			mm1.hpgSize = "2M"
+			// we are requesting 8Mi hugepages which again can be satisifed by numa zone 0
+			mm1.noOfhpgs = "8Mi"
+			testPod1 := mm1.createPodTemplate(profile, true, targetNode)
+			// Initialize test pod, check if the pod uses Numa node 0
+			err := initializePod(testPod1)
+			Expect(err).ToNot(HaveOccurred(), "Unable to initialize pod")
+			numaZone, err := GetMemoryNodes(testPod1, targetNode)
+			// Expect numa node 0 to be used by pod
+			Expect(numaZone).To(Equal("0"))
+			Expect(err).ToNot(HaveOccurred(), "Pod's numa affinity is %s instead of %s", numaZone, "0")
+			//Delete pod
+			err = mm1.removePod(testPod1)
+			Expect(err).ToNot(HaveOccurred())
+			//Schedule pod on numa zone 1
+			mm2.noOfhpgs = "4Gi"
+			mm2.memory = "200Mi"
+			mm2.hpgSize = profile.Spec.HugePages.Pages[1].Size
+			mm2.cpu = fmt.Sprintf("%d", len(available_node1_cpus)-2)
+			testPod2 := mm2.createPodTemplate(profile, true, targetNode)
+			// Initialize test pod, check if the pod uses Numa node 1
+			err = initializePod(testPod2)
+			Expect(err).ToNot(HaveOccurred(), "Unable to initialize pod")
+			numaZone, err = GetMemoryNodes(testPod2, targetNode)
+			// Expect numa node 1 to be used by pod
+			Expect(numaZone).To(Equal("1"))
+			Expect(err).ToNot(HaveOccurred(), "Pod's numa affinity is %s instead of %s", numaZone, "1")
+			//Delete pod
+			err = mm2.removePod(testPod2)
+			Expect(err).ToNot(HaveOccurred(), "failed to remove testpod")
+		})
+
+		AfterAll(func() {
+			By("Reverting the Profile")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			currentSpec, _ := json.Marshal(profile.Spec)
+			spec, _ := json.Marshal(initialProfile.Spec)
+			// revert only if the profile changes.
+			if !bytes.Equal(currentSpec, spec) {
+				spec, err := json.Marshal(initialProfile.Spec)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testclient.Client.Patch(context.TODO(), profile,
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+					),
+				)).ToNot(HaveOccurred())
+
+				By("Applying changes in performance profile and waiting until mcp will start updating")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+				By("Waiting when mcp finishes updates")
+				mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+			}
+		})
+	})
+})
+
+func checkPodEvent(testpod *corev1.Pod, podEventReason string) error {
+	podEvents, err := events.GetEventsForObject(testclient.Client, testpod.Namespace, testpod.Name, string(testpod.UID))
+	if err != nil {
+		testlog.Error(err)
+		return err
+	}
+	testlog.Infof("log pod %s/%s events due to failure", testpod.Namespace, testpod.Name)
```

Comment:
logging the events is not only due to failures but also the function might be called to check if the pod is "scheduled" like here:
https://github.com/openshift/cluster-node-tuning-operator/pull/573/files#diff-95351e9262524b4c12a711f6d5265bb323f57bdef3fc80aeb0e4d9d78d5e5c1bR824

Reply:
Addressed this in the latest commit.